src/Raven.Client/Documents/Session/StreamQueryStatistics.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Globalization;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Session
+{
+    public class StreamQueryStatistics
+    {
+        public string IndexName { get; set; }
+        public bool IsStale { get; set; }
+        public DateTime IndexTimestamp { get; set; }
+        public int TotalResults { get; set; }
+        public long ResultEtag { get; set; }
+
+        internal void UpdateQueryStats(string indexName, string isStale, string indexTimestamp, string totalResults, string resultEtag)
```

Comment:
I think you can safely move everything from here to EnsureValidStreamQueryStats, this is also a part of validation checking if the type of value is parsable to int for example